test/index.js

```diff
@@ -16,1680 +15,1339 @@ const internals = {};
 // Test shortcuts
 
 const lab = exports.lab = Lab.script();
-const describe = lab.describe;
-const it = lab.it;
-const expect = Code.expect;
+const { describe, it, expect } = lab;
+
+
+// Helpers
+const Helpers = require('./helpers');
 
 
 describe('scheme', () => {
 
-    it('fails with no plugin options', (done) => {
+    it('fails with no plugin options', () => {
 
         const server = new Hapi.Server();
-        server.connection();
-        server.register(require('../'), (err) => {
+        server.register(require('../'));
 
-            expect(err).to.not.exist();
+        expect(() => {
 
-            expect(() => {
+            server.auth.strategy('session', 'cookie', {});
+        }).to.throw(Error);
+    });
 
-                server.auth.strategy('default', 'cookie', true, {});
-            }).to.throw(Error);
+    it('passes with a password configured', () => {
 
-            done();
-        });
+        const server = new Hapi.Server();
+        server.register(require('../'));
+
+        expect(() => {
+
+            server.auth.strategy('session', 'cookie', { password: 'password-should-be-32-characters' });
+            server.auth.default('session');
+        }).to.not.throw();
     });
 
-    it('passes with a password configured', (done) => {
+    it('passes with a password configured which is a Buffer', () => {
 
         const server = new Hapi.Server();
-        server.connection();
-        server.register(require('../'), (err) => {
+        server.register(require('../'));
+
+        expect(() => {
 
-            expect(err).to.not.exist();
+            server.auth.strategy('session', 'cookie', { password: new Buffer('foobar') });
```

Comment:
`new Buffer` [is deprecated](https://nodejs.org/docs/latest-v8.x/api/buffer.html#buffer_new_buffer_string_encoding) since node v6. Should be replaced by `Buffer.from`.

Reply:
Good catch! Updated.